packages/aa/test/utils.js

```diff
@@ -2,6 +2,53 @@ const path = require('node:path')
 const { symlink } = require('node:fs/promises')
 const { existsSync } = require('node:fs')
 
+const translateHeapStats = (stats = []) => {
+  const result = {}
+  for (const { space_name, space_used_size } of stats) {
+    result[space_name] = space_used_size
+  }
+  return result
+}
+
+const updateMaxEachKey = (current, update) => {
+  for (const key in current) {
+    current[key] = Math.max(current[key], update[key])
+  }
+}
+
+const diffEachKey = (a, b) => {
+  const result = {}
+  for (const key in a) {
+    result[key] = b[key] - a[key]
+  }
+  return result
+}
+
+const toHumanReadable = (obj) => {
+  const result = {}
+  for (const key in obj) {
+    if (obj[key] > 0) result[key] = `+${(obj[key] / 1024 / 1024).toFixed(4)} MB`
+  }
+  return result
+}
+
+const recordMemorySpike = (frequency = 10) => {
+  const v8 = require('v8')
+  const initial = translateHeapStats(v8.getHeapSpaceStatistics())
+  const result = { ...initial }
+  const collect = () =>
+    updateMaxEachKey(result, translateHeapStats(v8.getHeapSpaceStatistics()))
+  const interval = setInterval(collect, frequency)
```

Comment:
```suggestion
  const interval = setInterval(collect, frequency).unref()
```

unless there's a reason not to do that, it will prevent future bugs encountered by interloping developers of the "why does my test hang???" variety

Reply:
yeah, why not.